drivers/gpio/gpio_rt1718s.h

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_GPIO_RT1718S_H_
+#define ZEPHYR_DRIVERS_GPIO_RT1718S_H_
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+
+#define RT1718S_GPIO_NUM 3
+
+#define RT1718S_REG_ALERT		       0x10
+#define RT1718S_REG_ALERT_VENDOR_DEFINED_ALERT BIT(15)
+
+#define RT1718S_REG_ALERT_MASK			    0x12
+#define RT1718S_REG_ALERT_MASK_VENDOR_DEFINED_ALERT BIT(15)
+
+#define RT1718S_REG_RT_MASK8	     0xA6
+#define RT1718S_REG_RT_MASK8_GPIO1_R BIT(0)
+#define RT1718S_REG_RT_MASK8_GPIO2_R BIT(1)
+#define RT1718S_REG_RT_MASK8_GPIO3_R BIT(2)
+#define RT1718S_REG_RT_MASK8_GPIO1_F BIT(4)
+#define RT1718S_REG_RT_MASK8_GPIO2_F BIT(5)
+#define RT1718S_REG_RT_MASK8_GPIO3_F BIT(6)
+
+#define RT1718S_REG_RT_INT8	    0xA8
+#define RT1718S_REG_RT_INT8_GPIO1_R BIT(0)
+#define RT1718S_REG_RT_INT8_GPIO2_R BIT(1)
+#define RT1718S_REG_RT_INT8_GPIO3_R BIT(2)
+#define RT1718S_REG_RT_INT8_GPIO1_F BIT(4)
+#define RT1718S_REG_RT_INT8_GPIO2_F BIT(5)
+#define RT1718S_REG_RT_INT8_GPIO3_F BIT(6)
+#define RT1718S_GPIO_INT_MASK                                                                      \
+	(RT1718S_REG_RT_INT8_GPIO1_R | RT1718S_REG_RT_INT8_GPIO2_R | RT1718S_REG_RT_INT8_GPIO3_R | \
+	 RT1718S_REG_RT_INT8_GPIO1_F | RT1718S_REG_RT_INT8_GPIO2_F | RT1718S_REG_RT_INT8_GPIO3_F)
+
+#define RT1718S_REG_RT_ST8	   0xAA
+#define RT1718S_REG_RT_ST8_GPIO1_I BIT(0)
+#define RT1718S_REG_RT_ST8_GPIO2_I BIT(1)
+#define RT1718S_REG_RT_ST8_GPIO3_I BIT(2)
+
+#define RT1718S_REG_GPIO_CTRL(pin) (0xED + pin)
+#define RT1718S_REG_GPIO_CTRL_PU   BIT(5)
+#define RT1718S_REG_GPIO_CTRL_PD   BIT(4)
+#define RT1718S_REG_GPIO_CTRL_OD_N BIT(3)
+#define RT1718S_REG_GPIO_CTRL_OE   BIT(2)
+#define RT1718S_REG_GPIO_CTRL_O	   BIT(1)
+#define RT1718S_REG_GPIO_CTRL_I	   BIT(0)
+
+/* RT1718S chip driver config */
+struct rt1718s_config {
+	/* I2C device */
+	const struct i2c_dt_spec i2c_dev;
+	/* Alert GPIO pin */
+	const struct gpio_dt_spec irq_gpio;
```

Comment:
Nit - you can guard this with CONFIG_GPIO_RT1718S_INTERRUPT.

Reply:
I've considered this, but it requires changing `if (IS_ENABLED(CONFIG_GPIO_RT1718S_INTERRUPT)) {` condition to `#ifdef` statement, otherwise the compiler complains `'const struct rt1718s_config' has no member named 'irq_gpio'` inside the if statement. I'm not sure if it is worth to change that. Is there another way how to handle it that I'm not aware of?